test/common/unit/net/test_bolt1.py

```diff
@@ -0,0 +1,385 @@
+    # Init message requires global_features(FeatureVector), local_features (FeatureVector) and optionally a NetworksTLV
+    gf = FeatureVector.from_bytes(b"\x02")
+    lf = FeatureVector()
+    im = InitMessage(gf, lf)
+    assert isinstance(im, InitMessage)
+    assert im.global_features == gf and im.local_features == lf and im.networks is None
+
+    # Same with networks
+    networks = NetworksTLV([get_random_value_hex(32) for _ in range(5)])
+    im2 = InitMessage(gf, lf, networks)
+    assert isinstance(im2, InitMessage)
+    assert im2.global_features == gf and im2.local_features == lf and im2.networks is networks
+
+
+def test_init_message_wrong():
+    # No FeatureVectors
+    with pytest.raises(TypeError, match="global_features and local_features must be FeatureVector instances"):
+        InitMessage("features", FeatureVector())
+    with pytest.raises(TypeError, match="global_features and local_features must be FeatureVector instances"):
+        InitMessage(FeatureVector(), "features")
+
+    # TLV must be NetworksTLV is fet (for now)
+    with pytest.raises(TypeError, match="networks must be of type NetworksTLV"):
+        InitMessage(FeatureVector(), FeatureVector(), "TLV")
+
+
+def test_init_message_from_bytes():
+    # Message type must be init and size at least 6 (type + gflen + flen)
+    mtype = b"\x00\x10"
+    gflen = b"\x00\x00"
+    flen = gflen
+    im = InitMessage.from_bytes(mtype + gflen + flen)
+    assert (
+        isinstance(im, InitMessage)
+        and im.type == mtype
+        and im.global_features.serialize() == im.local_features.serialize() == b""
+    )
+
+    # A more meaningful init (with some features)
+    mtype = b"\x00\x10"
+    global_features = b"\x2a\xaa\xaa"  # All odd
+    gflen = b"\x00\x03"
+    local_features = b"\x01"  # Feature 1 even
+    flen = b"\x00\x01"
+    im2 = InitMessage.from_bytes(mtype + gflen + global_features + flen + local_features)
+    assert (
+        isinstance(im2, InitMessage)
+        and im2.type == mtype
+        and im2.global_features.serialize() == global_features
+        and im2.local_features.serialize() == local_features
+    )
+
+    # With some networks
+    networks = NetworksTLV([get_random_value_hex(32) for _ in range(5)])
+    im3 = InitMessage.from_bytes(mtype + gflen + global_features + flen + local_features + networks.serialize())
+    assert (
+        isinstance(im3, InitMessage)
+        and im3.type == mtype
+        and im3.global_features.serialize() == global_features
+        and im3.local_features.serialize() == local_features
+        and im3.networks == networks
+    )
+
+
+def test_init_message_from_bytes_wrong():
+    # Message is not bytes
+    with pytest.raises(TypeError, match="message be must a bytearray"):
+        InitMessage.from_bytes("message")
+
+    # Message is not long enough < 6
+    with pytest.raises(ValueError, match="message be must at least 6-byte long"):
+        InitMessage.from_bytes(b"\x00\x10\x00")
```

Comment:
Since the test is for < 6, I thing `from_bytes` should be given exactly 5 bytes here.